From 24d426921b509b180f22967f4e88dc21ea7c233e Mon Sep 17 00:00:00 2001
Message-Id: <24d426921b509b180f22967f4e88dc21ea7c233e.1297957056.git.jdenemar@redhat.com>
From: Eric Blake <eblake@redhat.com>
Date: Thu, 17 Feb 2011 16:15:39 +0800
Subject: [PATCH] build: fix parted detection at configure time

* configure.ac (PARTED_FOUND): Issue configure error if
--with-storage-disk=yes but no parted is found.
(cherry picked from commit 59b34f139d85dd11fa380833e47ea9784b3e2e39)

cherry-picked as a dependency for
    https://bugzilla.redhat.com/show_bug.cgi?id=611443
for the following 2 reasons:

  1) It just corrects the wrong logic in configure.ac, and makes
     sense for RHEL6.1.
  2) It prevents cherry-picking patches which also hack on configure.ac,
     unless to modify the patches.
---
 configure.ac |   15 +++++++++------
 1 files changed, 9 insertions(+), 6 deletions(-)

diff --git a/configure.ac b/configure.ac
index 19e9588..765cc80 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1722,17 +1722,18 @@ AC_SUBST([DEVMAPPER_LIBS])
 
 LIBPARTED_CFLAGS=
 LIBPARTED_LIBS=
-if test "$with_storage_disk" = "yes" || test "$with_storage_disk" = "check"; then
+if test "$with_storage_disk" = "yes" ||
+   test "$with_storage_disk" = "check"; then
   AC_PATH_PROG([PARTED], [parted], [], [$PATH:/sbin:/usr/sbin])
   if test -z "$PARTED" ; then
-    with_storage_disk=no
     PARTED_FOUND=no
   else
     PARTED_FOUND=yes
   fi
 
-  if test "$with_storage_disk" != "no" && test "x$PKG_CONFIG" != "x" ; then
-    PKG_CHECK_MODULES(LIBPARTED, libparted >= $PARTED_REQUIRED, [], [PARTED_FOUND=no])
+  if test "$PARTED_FOUND" = "yes" && test "x$PKG_CONFIG" != "x" ; then
+    PKG_CHECK_MODULES([LIBPARTED], [libparted >= $PARTED_REQUIRED], [],
+      [PARTED_FOUND=no])
   fi
   if test "$PARTED_FOUND" = "no"; then
     # RHEL-5 vintage parted is missing pkg-config files
@@ -1758,8 +1759,10 @@ if test "$with_storage_disk" = "yes" || test "$with_storage_disk" = "check"; the
   fi
 
   if test "$with_storage_disk" = "yes"; then
-    AC_DEFINE_UNQUOTED([WITH_STORAGE_DISK], 1, [whether Disk backend for storage driver is enabled])
-    AC_DEFINE_UNQUOTED([PARTED],["$PARTED"], [Location or name of the parted program])
+    AC_DEFINE_UNQUOTED([WITH_STORAGE_DISK], 1,
+      [whether Disk backend for storage driver is enabled])
+    AC_DEFINE_UNQUOTED([PARTED],["$PARTED"],
+      [Location or name of the parted program])
   fi
 fi
 AM_CONDITIONAL([WITH_STORAGE_DISK], [test "$with_storage_disk" = "yes"])
-- 
1.7.4.1

